crates/klave/src/crypto/subtle.rs

```diff
@@ -183,13 +183,13 @@ impl From<RsaHashedKeyGenParams> for GenAlgorithm {
 
 #[derive(Deserialize, Serialize)]
 pub enum EncryptAlgorithm {
-    RsaOaepEncryptionMetadata(RsaOaepEncryptionMetadata),
+    RsaOaepParams(RsaOaepParams),
     AesGcmParams(AesGcmParams),
 }
 
-impl From<RsaOaepEncryptionMetadata> for EncryptAlgorithm {
-    fn from(params: RsaOaepEncryptionMetadata) -> Self {
-        EncryptAlgorithm::RsaOaepEncryptionMetadata(params)
+impl From<RsaOaepParams> for EncryptAlgorithm {
```

Comment:
I'm not convinced by this trait implementation (and of course same for line 227). `From` and `Into` are supposed to be used for *conversions*. Here this looks more like a kind of constructor, with little added value : is `my_params.into()` really better than an explicit `EncryptAlgorithm::RsaOaepParams(my_params)` ?

OTOH I'm not sure the variants of the enum are properly named : shouldn't they reflect the *algorithm*, with the param only as the payload ? i.e. :
```
pub enum EncryptAlgorithm {
    RsaOaep(RsaOaepParams),
    AesGcm(AesGcmParams),
}
```

Reply:
Funnily enough, I couldn't find where the removal of the `From` traits had impact (where `::from()` or `.into()` would have been used).